lib/std/process/Child.zig

```diff
@@ -847,9 +853,19 @@ fn spawnWindows(self: *ChildProcess) SpawnError!void {
     defer if (cwd_w) |cwd| self.allocator.free(cwd);
     const cwd_w_ptr = if (cwd_w) |cwd| cwd.ptr else null;
 
-    const maybe_envp_buf = if (self.env_map) |env_map| try process.createWindowsEnvBlock(self.allocator, env_map) else null;
-    defer if (maybe_envp_buf) |envp_buf| self.allocator.free(envp_buf);
-    const envp_ptr = if (maybe_envp_buf) |envp_buf| envp_buf.ptr else null;
+    const maybe_envp_buf = m: {
+        const prog_fd: ?usize = if (prog_pipe_wr) |h| @intFromPtr(h) else null;
+        if (self.env_map) |env_map| {
+            break :m try process.createWindowsEnvBlock(self.allocator, env_map, .{ .zig_progress_fd = prog_fd });
+        } else {
+            const env_map = try process.getEnvMap(self.allocator);
+            break :m try process.createWindowsEnvBlock(self.allocator, &env_map, .{ .zig_progress_fd = prog_fd });
+        }
+    };
+    // defer if (maybe_envp_buf) |envp_buf| self.allocator.free(envp_buf);
```

Comment:
None of the cases above return null, so "maybe_envp_buf" ends up not being an optional...
I guess as an optimization we could return null here if prog_pipe_wr/prog_fd is null